### ISSUE TITLE
[Linux] radios and switches have wrong accessibility roles

### ISSUE THREAD

**Issue by jpnurmi** [flutter/flutter#101434]
Orca Screen Reader considers radios and switches as checkboxes because anything checkable or toggleable has the "checkbox" accessibility role.

## Steps to Reproduce

1. Turn on the screen reader (Settings -> Accessibility -> Screen Reader) 
2. Execute `flutter run` on the code sample 
3. Press _Tab_ to navigate around 

**Expected results:** 

Match the behavior with GTK:

- Radio buttons are read as "radio buttons"
- Switches are read as "toggle buttons"

**Actual results:** 

Checkboxes, radio buttons, and switches are all read as "checkboxes".

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  final checked = ValueNotifier<bool>(true);

  runApp(
    MaterialApp(
      home: Scaffold(
        body: ValueListenableBuilder<bool>(
          valueListenable: checked,
          builder: (context, value, _) {
            return Column(
              children: [
                Checkbox(
                  value: value,
                  onChanged: (v) => checked.value = v!,
                ),
                Radio<bool>(
                  value: value,
                  groupValue: true,
                  onChanged: (v) => checked.value = v!,
                ),
                Switch(
                  value: value,
                  onChanged: (v) => checked.value = v,
                )
              ],
            );
          },
        ),
      ),
    ),
  );
}
```

</details>

<details>
  <summary>Logs</summary>

```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.393, on Ubuntu Jammy Jellyfish (development branch) 5.15.0-25-generic, locale en_US.UTF-8)
    • Flutter version 2.13.0-0.0.pre.393 at /home/jpnurmi/Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 94fefaa49d (5 hours ago), 2022-04-06 17:17:34 +0530
    • Engine revision 45bedbe841
    • Dart version 2.17.0 (build 2.17.0-266.0.dev)
    • DevTools version 2.12.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/jpnurmi/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_SDK_ROOT = /home/jpnurmi/Android/Sdk
    • Java binary at: /home/jpnurmi/Android/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Linux toolchain - develop for Linux desktop
    • Ubuntu clang version 14.0.0-1ubuntu1
    • cmake version 3.22.1
    • ninja version 1.10.1
    • pkg-config version 0.29.2

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/jpnurmi/Android/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code
    • VS Code at /snap/code/current
    • Flutter extension version 3.38.1

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Ubuntu Jammy Jellyfish (development branch) 5.15.0-25-generic

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by jpnurmi** [comment]
Fix available: https://github.com/flutter/engine/pull/32454

**Comment by maheshmnj** [comment]
Hi @jpnurmi, thanks for filing the issue. I ran your code sample with a screen reader enabled (orca screen reader) on the latest stable and the master channel. But the screen Reader doesn't read anything at all. Tried changing the state of widgets with the mouse and keyboard. The screen reader only mentions the keys pressed, But nothing about the widget or its state. Verified on the Desktop and the web.


<details>
<summary>flutter doctor -v (Linux)</summary>

```
[✓] Flutter (Channel stable, 2.10.4, on Ubuntu 20.04.4 LTS 5.13.0-30-generic, locale en_GB.UTF-8)
    • Flutter version 2.10.4 at /home/mahesh/Development/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (12 days ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/mahesh/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio
    • Flutter plugin version 63.2.1
    • Dart plugin version 203.8452
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.64.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.38.1

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 20.04.4 LTS 5.13.0-30-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 98.0.4758.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.392, on Ubuntu 20.04.4 LTS 5.13.0-30-generic, locale en_GB.UTF-8)
    • Flutter version 2.13.0-0.0.pre.392 at /home/mahesh/Development/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 120b3deb18 (20 minutes ago), 2022-04-06 10:59:37 +0200
    • Engine revision 45bedbe841
    • Dart version 2.17.0 (build 2.17.0-266.0.dev)
    • DevTools version 2.12.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/mahesh/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio
    • Flutter plugin version 63.2.1
    • Dart plugin version 203.8452
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.64.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.38.1

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 20.04.4 LTS 5.13.0-30-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 98.0.4758.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>  

Let me know if I am missing anything?

**Comment by jpnurmi** [comment]
There might be several issues in the game (*) but I'm trying to focus on the button roles in this bug report. :smile: Here's a recording with the latest Flutter master on Ubuntu 22.04.

https://user-images.githubusercontent.com/140617/162153311-61abe9f4-fa98-41a5-a227-04428df4b3ef.mp4

*) The Orca screen reader tends to have strange issues with Flutter apps. Sometimes it gets in a state that it starts somehow ignoring Flutter apps/widget even though other non-Flutter apps are read fine.

**Comment by maheshmnj** [comment]
I see, On my machine orca doesn't seem to read the futter widgets at all
### output from stable 
https://user-images.githubusercontent.com/31410839/162206570-e31f0670-2a55-4c2d-9e7c-15055581b6f6.mp4

While on the master channel it only reads "Tab" I am using orca `3.36.2`

Labeling this issue for further insights from the team.

**Comment by maheshmnj** [comment]
Update: I modified orca settings to read components under the mouse and  I can see that all three widgets are read as checkboxes.

*Unmute the video for audiovisual output*

https://user-images.githubusercontent.com/31410839/162213449-9424c2e4-127b-4451-9ea0-5921202782ee.mp4

<details>
<summary>flutter doctor -v (Linux)</summary>

```
[✓] Flutter (Channel stable, 2.10.4, on Ubuntu 20.04.4 LTS 5.13.0-30-generic, locale en_GB.UTF-8)
    • Flutter version 2.10.4 at /home/mahesh/Development/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c860cba910 (12 days ago), 2022-03-25 00:23:12 -0500
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/mahesh/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio
    • Flutter plugin version 63.2.1
    • Dart plugin version 203.8452
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.64.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.38.1

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 20.04.4 LTS 5.13.0-30-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 98.0.4758.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
```
[✓] Flutter (Channel master, 2.13.0-0.0.pre.392, on Ubuntu 20.04.4 LTS 5.13.0-30-generic, locale en_GB.UTF-8)
    • Flutter version 2.13.0-0.0.pre.392 at /home/mahesh/Development/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 120b3deb18 (20 minutes ago), 2022-04-06 10:59:37 +0200
    • Engine revision 45bedbe841
    • Dart version 2.17.0 (build 2.17.0-266.0.dev)
    • DevTools version 2.12.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/mahesh/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2020.3)
    • Android Studio at /home/mahesh/Downloads/android-studio-2020.3.1.25-linux/android-studio
    • Flutter plugin version 63.2.1
    • Dart plugin version 203.8452
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.64.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.38.1

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 20.04.4 LTS 5.13.0-30-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 98.0.4758.102

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>
